src/farbfeld.rs

```diff
@@ -101,6 +91,76 @@ impl<R: Read> Read for FarbfeldReader<R> {
     }
 }
 
+impl<R: Read + Seek> Seek for FarbfeldReader<R> {
+    fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
+        fn checked_seek<F: FnOnce(u64) -> io::Result<()>>(from: u64, by: i64, cbk: F) -> io::Result<bool> {
+            match if by < 0 {
+                from.checked_sub(by.wrapping_neg() as u64)
+            } else {
+                from.checked_add(by as u64)
+            } {
+                Some(off) if off < ORIGIN => Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid seek to a negative position")),
+                Some(off) => {
+                    cbk(off)?;
+                    Ok(off % 2 == 1)
+                }
+                None => Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid seek to a negative or overflowing position")),
+            }
+        }
+
+        /// Start of pixel data (magic + width + height)
+        static ORIGIN: u64 = 8 + 4 + 4;
+
+        let mut first_seek = 0;
```

Comment:
This applies to multiple uses of `first_seek`, so adding it here.

The offset returned from `seek` should be relative to the image stream. That is, one should be able to use this offset in another `seek(SeekFrom::Start(off))` to return to the same position. In particular this assumption is used in the implementation of `Seek::stream_position`. However, the return value here is always a mix of the offset of the inner stream offset by a static `ORIGIN`. This is incorrect if the beginning of the image in the inner stream is not `0`, which may happen if the image is read from an archive.

This may also be good to cover in a test case.

Reply:
Haven't thought of that before, but I rewrote it and it works with mid-archive streams now